api/app/Jobs/UpdateUserWeather.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\User;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Redis;
+use Illuminate\Support\Facades\Log;
+
+class UpdateUserWeather implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    protected $user;
+    protected const REQUEST_TIMEOUT = 0.5; // 500ms timeout
+    protected const REDIS_PREFIX = 'weather:';
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(User $user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        try {
+            $response = Http::timeout(self::REQUEST_TIMEOUT)
+                ->get("https://api.openweathermap.org/data/2.5/weather", [
```

Comment:
```suggestion
                ->get(config('weather.base_url'), [
```
You can use your config here as well